internal/auth/authController.go

```diff
@@ -0,0 +1,74 @@
+package auth
+
+import (
+	"eventapi/internal/database"
+	"eventapi/internal/session"
+	"fmt"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+const userInfoURL = "https://www.googleapis.com/oauth2/v2/userinfo"
+
+var us *database.UserService
+var ss *session.SessionService
+
+func init() {
+	us = database.NewUserService()
+	ss = session.NewSessionService()
+}
+
+// Login handler
+func Login(c *fiber.Ctx) error {
+	url := getEntryPoint()
+	return c.Redirect(url)
+}
+
+// oauth callback
+func Callback(c *fiber.Ctx) error {
+	token, err := exchange(c.Context(), c.FormValue("state"), c.FormValue("code"))
+	if err != nil {
+		return err
+	}
+
+	u, err := getUserInfo(token.AccessToken)
+
+	if err != nil {
+		return fmt.Errorf("could not fetch user info: %w", err)
```

Comment:
nit: would switch the wording to: `error fetching user info`

`error <doing whatever>: <detail>`